File: tips/enums/enum.swift
enum DeliveryStatus {
    case pending
    case inTransit
    case delivered
    case cancelled

    var message: String {
        switch self {
        case .pending:
            return "Tu paquete está siendo preparado 📦"
        case .inTransit:
            return "¡Va en camino! 🚚"
        case .delivered:
            return "Entregado con éxito ✅"
        case .cancelled:
            return "Envío cancelado ❌"
        }
    }

    var isFinalized: Bool {
        return self == .delivered || self == .cancelled
    }
}

let status: DeliveryStatus = DeliveryStatus.delivered
print(status.message)
print(status.isFinalized)

File: tips/enums/enum.kt
enum class DeliveryStatus {
    PENDING,
    IN_TRANSIT,
    DELIVERED,
    CANCELLED;

    val message: String
        get() = when (this) {
            PENDING -> "Tu paquete está siendo preparado 📦"
            IN_TRANSIT -> "¡Va en camino! 🚚"
            DELIVERED -> "Entregado con éxito ✅"
            CANCELLED -> "Envío cancelado ❌"
        }

    val isFinalized: Boolean
        get() = this == DELIVERED || this == CANCELLED
}

fun main() {
    val status = DeliveryStatus.DELIVERED
    println(status.message)
    println(status.isFinalized)
}